packages/tools/src/eventListeners/touch/touchStartListener.ts

```diff
@@ -290,126 +333,128 @@ function _onTouchDrag(evt: TouchEvent) {
-      startPointsList: copyPointsList(state.startPointsList),
-      lastPointsList: copyPointsList(lastPointsList),
-      currentPointsList,
-      startPoints: getMeanTouchPoints(state.startPointsList),
-      lastPoints: getMeanTouchPoints(lastPointsList),
-      currentPoints: getMeanTouchPoints(currentPointsList),
-      deltaPoints,
-      deltaDistance,
-    };
-
-    triggerEventCallback(eventDetail.element, TOUCH_END, eventDetail);
-    state = JSON.parse(JSON.stringify(defaultState));
-
-    // Remove our temporary handlers which is only added when normalized touch
-    // start fires
-    document.removeEventListener('touchmove', _onTouchDrag);
-  } else {
-    // a tap occured
-    _onTouchTap(evt);
-  }
-
-  document.removeEventListener('touchend', _onTouchEnd);
-}
-
-function _onTouchTap(evt: TouchEvent): void {
   const currentPointsList = getTouchEventPoints(evt, state.element);
   const lastPointsList = _updateTouchEventsLastPoints(
     state.element,
     state.lastPointsList
   );
+  const deltaPoints =
+    currentPointsList.length === lastPointsList.length
+      ? getDeltaPoints(currentPointsList, lastPointsList)
+      : getDeltaPoints(currentPointsList, currentPointsList);
+  const deltaDistance =
+    currentPointsList.length === lastPointsList.length
+      ? getDeltaDistanceBetweenIPoints(currentPointsList, lastPointsList)
+      : getDeltaDistanceBetweenIPoints(currentPointsList, currentPointsList);
+  const eventDetail: EventTypes.TouchEndEventDetail = {
+    event: evt,
+    eventName: TOUCH_END,
+    element: state.element,
+    renderingEngineId: state.renderingEngineId,
+    viewportId: state.viewportId,
+    camera: {},
+    startPointsList: copyPointsList(state.startPointsList),
+    lastPointsList: copyPointsList(lastPointsList),
+    currentPointsList,
+    startPoints: getMeanTouchPoints(state.startPointsList),
+    lastPoints: getMeanTouchPoints(lastPointsList),
+    currentPoints: getMeanTouchPoints(currentPointsList),
+    deltaPoints,
+    deltaDistance,
+  };
+
+  triggerEventCallback(eventDetail.element, TOUCH_END, eventDetail);
+  _checkTouchTap(evt);
 
+  // reset to default state
+  state = JSON.parse(JSON.stringify(defaultState));
+  document.removeEventListener('touchmove', _onTouchDrag);
+  document.removeEventListener('touchend', _onTouchEnd);
+}
+
+function _checkTouchTap(evt: TouchEvent): void {
+  const currentTime = new Date().getTime();
+  const startTime = state.startTime.getTime();
+  if (currentTime - startTime >= tapState.tapWindow) return;
+
+  // first tap, initialize the state
+  if (tapState.taps === 0) {
+    tapState.element = state.element;
+    tapState.renderingEngineId = state.renderingEngineId;
+    tapState.viewportId = state.viewportId;
+    tapState.startPointsList = state.startPointsList;
+    tapState.startTime = new Date();
+  }
+
+  // check that taps are within interval
+  if (currentTime - tapState.startTime.getTime() >= tapState.tapWindow) return;
```

Comment:
I think it might be best to use a `>` here instead of `>=` so that we have a true tolerance and not tolerance - 1.

Reply:
We can change to >

Reply:
👍